components/prism-yaml.js

```diff
@@ -16,32 +16,32 @@ Prism.languages.yaml = {
 		alias: 'important'
 	},
 	'datetime': {
-		pattern: /([:\-,[{]\s*(?:![^\s]+)?[ \t]*)(?:\d{4}-\d\d?-\d\d?(?:[tT]|[ \t]+)\d\d?:\d{2}:\d{2}(?:\.\d*)?[ \t]*(?:Z|[-+]\d\d?(?::\d{2})?)?|\d{4}-\d{2}-\d{2}|\d\d?:\d{2}(?::\d{2}(?:\.\d*)?)?)(?=[ \t]*(?:$|,|]|}))/m,
+		pattern: /([:\-,[{]\s*(?:![^\s]+)?(?:[&*][^\s\[\]{},]+)?[ \t]*)(?:\d{4}-\d\d?-\d\d?(?:[tT]|[ \t]+)\d\d?:\d{2}:\d{2}(?:\.\d*)?[ \t]*(?:Z|[-+]\d\d?(?::\d{2})?)?|\d{4}-\d{2}-\d{2}|\d\d?:\d{2}(?::\d{2}(?:\.\d*)?)?)(?=[ \t]*(?:$|,|]|}))/m,
 		lookbehind: true,
 		alias: 'number'
 	},
 	'boolean': {
-		pattern: /([:\-,[{]\s*(?:![^\s]+)?[ \t]*)(?:true|false)[ \t]*(?=$|,|]|})/im,
+		pattern: /([:\-,[{]\s*(?:![^\s]+)?(?:[&*][^\s\[\]{},]+)?[ \t]*)(?:true|false)[ \t]*(?=$|,|]|})/im,
 		lookbehind: true,
 		alias: 'important'
 	},
 	'null': {
-		pattern: /([:\-,[{]\s*(?:![^\s]+)?[ \t]*)(?:null|~)[ \t]*(?=$|,|]|})/im,
+		pattern: /([:\-,[{]\s*(?:![^\s]+)?(?:[&*][^\s\[\]{},]+)?[ \t]*)(?:null|~)[ \t]*(?=$|,|]|})/im,
```

Comment:
I don't really use YAML, so please help me here: The `(?:![^\s]+)?` is supposed to match [tags](http://blogs.perl.org/users/tinita/2018/01/introduction-to-yaml-schemas-and-tags.html), right?

So I have a few questions:

1. Is this valid: `foo: !!tag &anchor null` ? (not matched right now)
2. Is this valid: `foo: !!tag,&,an{}chor null` ? (matched)
3. Does this contain an anchor: `foo: !!tag&anchor null` ? (matched)

Nr. 2 and 3 are matched because the `[^\s]+` will consume the "anchor".

Reply:
1. Yes, I'm working on it.
2. This is not an anchor (and not matched as such). If this is a valid tag, everything's fine. If not, fixing tag matching is out-of-scope for this PR.
3. The same as 2.

Reply:
> If not, fixing tag matching is out-of-scope for this PR.

Fair enough. I looked it up in the spec, and yes, it's a bug. I'll make a PR for this issue after we merge this.

Regarding 1.:
Just make it this here. It'll work.

```suggestion
		pattern: /([:\-,[{]\s*(?:![^\s]+[ \t]*)?(?:[&*][^\s\[\]{},]+[ \t]*)?)(?:null|~)[ \t]*(?=$|,|]|})/im,
```